mqtt_client.c

```diff
@@ -744,8 +745,11 @@ static void esp_mqtt_task(void *pv)
                     client->state = MQTT_STATE_INIT;
                     client->reconnect_tick = platform_tick_get_ms();
                     ESP_LOGD(TAG, "Reconnecting...");
+                    break;
                 }
-                vTaskDelay(client->wait_timeout_ms / 2 / portTICK_RATE_MS);
+                xEventGroupClearBits(client->status_bits, RECONNECT_BIT);
```

Comment:
What is the purpose of clearing bits here? Wouldn't we miss reconnect command if it executed other than while waiting in the below EventWaitBits?

I know that the `esp-mqtt` is not yet thread safe, but what about typical call of `esp_mqtt_client_reconnect` on disconnect event? My suspicion is that it would be missed. Would not it make more sense to put it under `MQTT_STATE_INIT` ?

Reply:
Indeed clearing RECONNECT_BIT here looks fishy. I've been running with this patch applied locally for quite some time so can't remember why I decided to clear it here. I think I just needed the bit to be cleared somewhere for subsequent reconnects to function properly.

Agree that it makes more sense to put it under MQTT_STATE_INIT instead. Will upload an updated version shortly.